boilermaker/retries.py

```diff
@@ -95,27 +115,35 @@ def inc(self, when: datetime.datetime):
 
 
 class RetryException(Exception):
-    def __init__(self, msg: str, policy: RetryPolicy):
+    def __init__(self, msg: str, policy: RetryPolicy | None = None):
```

Comment:
I do think it's confusing that the retry exception can contain a new policy

Reply:
Because of these exceptions being the control-flow, it's hard to change how retries happen without allowing something like this.